pkgs/development/libraries/science/math/magma/generic.nix

```diff
@@ -119,15 +124,14 @@ stdenv.mkDerivation {
     cmake
     ninja
     gfortran
-  ];
+  ] ++ lists.optional cudaSupport cuda-native-redist;
 
   buildInputs = [
     libpthreadstubs
     lapack
     blas
-  ] ++ lists.optionals cudaSupport [
-    cuda_joined
-  ] ++ lists.optionals rocmSupport [
+  ] ++ lists.optional cudaSupport cuda-redist
```

Comment:
please prefer optionals. See https://discourse.nixos.org/t/lib-optional-considered-harmful/7081 for more info. The fact that optional exists at all is a bit unfortunate.

Reply:
also let's try to minimize our impact on the git blame by only touching line 129 if possible